### PR TITLE
Plugins: Add enablePluginsTracingByDefault feature flag

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Adds `enablePluginsTracingByDefault` feature flag. When the feature flag is enabled and tracing is enabled on the Grafana instance, tracing will be enabled for all external plugins.

Normally, tracing for external plugins is opt-in via plugin settings, like so:

```ini
[tracing.opentelemetry.otlp]
address = 127.0.0.1:4317
propagation = jaeger,w3c

[plugin.abc]
tracing = true

[plugin.xyz]
tracing = true
```

when the feature flag is enabled, only the `tracing.opentelemetry.*` section is required:

:point_down: 

```ini
[tracing.opentelemetry.otlp]
address = 127.0.0.1:4317
propagation = jaeger,w3c

[feature_toggles]
enablePluginsTracingByDefault = true
```

This will allow to easily roll-out tracing for external plugins on per-instance basis in cloud.

**Why do we need this feature?**

Rolling out tracing for external plugins in cloud.

**Who is this feature for?**

Grafana cloud users and plugins platform.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/services/featuremgmt/registry.go [hunk 1]
```diff
@@ -1297,5 +1297,13 @@ var (
 			RequiresRestart: true,
 			Created:         time.Date(2024, time.January, 3, 12, 0, 0, 0, time.UTC),
 		},
+		{
+			Name:         "enablePluginsTracingByDefault",
+			Description:  "Enable plugin tracing for all external plugins",
+			FrontendOnly: false,
+			Stage:        FeatureStageExperimental,
+			Owner:        grafanaPluginsPlatformSquad,
+			Created:      time.Date(2024, time.January, 9, 12, 0, 0, 0, time.UTC),
```
Comment (wbrowne) [inline pkg/services/featuremgmt/registry.go:1306]: Not sure it's really necessary, but should we be setting `RequiresRestart` to `true` here?

File: pkg/plugins/envvars/envvars_test.go [hunk 2]
```diff
@@ -353,6 +353,30 @@ func TestInitializer_tracingEnvironmentVariables(t *testing.T) {
...
+					OpenTelemetry: defaultOTelCfg,
+				},
+				PluginSettings: map[string]map[string]string{pluginID: {}},
+				Features:       featuremgmt.WithFeatures(featuremgmt.FlagEnablePluginsTracingByDefault),
+			},
+			plugin: defaultPlugin,
+			exp:    expDefaultOtlp,
+		},
+		{
+			name: "enabled on plugin with wrong plugin setting but with enablePluginsTracingByDefault feature flag",
```
Comment (wbrowne) [inline pkg/plugins/envvars/envvars_test.go:369]: Why is it referred to as "wrong" plugin setting?

File: pkg/services/featuremgmt/registry.go (hunk 1 above)
Comment (xnyo) [reply pkg/services/featuremgmt/registry.go]: Good spot, I have added it :+1:

**Review by wbrowne — APPROVED** [review]
LGTM!

**Pull request merged**